### PR TITLE
Fire education - release

### PR DESCRIPTION
**Pull request by cmonfortep**
Task/Issue URL: https://app.asana.com/0/488551667048375/1199361818043729/f
Tech Design URL: 
CC: 

**Description**:
This PR introduces necessary changes to release Fire Button education to all users. We have also introduced some changes coming from our last product review.

See notes in https://app.asana.com/0/488551667048375/1199361818043729/f

**Steps to test this PR**:
> Common scenario
1. Fresh install
1. Visit any site with trackers
1. Dismiss Dax CTA
1. Ensure Fire pulsing animation is active
1. Click on the Fire button
1. Ensure CTA appears on top of clear data options

> User opens more than 1 tabs
1. Fresh install
1. perform a search
1. Dismiss Dax CTA
1. open a new tab
1. Visit a site with trackers
1. Dismiss CTA
1. Ensure Fire pulsing animation does not appear
1. Click on the Fire button
1. Ensure CTA does not appear

> User opens more than 1 tabs after pulsing animation started
1. Fresh install
1. Visit any site with trackers
1. Dismiss Dax CTA
1. Ensure Fire pulsing animation is active
1. Open a new tab
1. Ensure pulsing animation is not active
1. Click on the Fire button
1. Ensure CTA does not appear

> Database migration
1. Install develop
1. Start the onboarding flow
1. Install this branch
1. Open the app, ensure onboarding was completed

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: app/src/androidTest/java/com/duckduckgo/app/browser/BrowserTabViewModelTest.kt [hunk 1]
```diff
@@ -2996,6 +3001,24 @@ class BrowserTabViewModelTest {
         assertTrue(command.headers.isEmpty())
     }
 
+    @Test
+    fun whenFirePulsingAnimationStartsThenItStopsAfterOneHour() = coroutineRule.runBlocking {
```
Comment (cmonfortep) [inline app/src/androidTest/java/com/duckduckgo/app/browser/BrowserTabViewModelTest.kt:3005]: This test was missing and it was requested on a previous review.

File: app/src/main/java/com/duckduckgo/app/global/db/AppDatabase.kt [hunk 2]
```diff
@@ -372,6 +372,12 @@ class MigrationsProvider(
         }
     }
 
+    val MIGRATION_28_TO_29: Migration = object : Migration(28, 29) {
+        override fun migrate(database: SupportSQLiteDatabase) {
+            database.execSQL("UPDATE $USER_STAGE_TABLE_NAME SET appStage = \"${AppStage.ESTABLISHED}\" WHERE appStage = \"${AppStage.DAX_ONBOARDING}\"")
```
Comment (cmonfortep) [inline app/src/main/java/com/duckduckgo/app/global/db/AppDatabase.kt:377]: As per https://app.asana.com/0/0/1199221440116910/f

**Review by marcosholgado — CHANGES_REQUESTED** [review]
Nice job! Looks pretty good, just a few minor comments plus what we talked about on zoom regarding that flow :)

File: app/src/androidTest/java/com/duckduckgo/app/cta/ui/CtaViewModelTest.kt [hunk 3]
```diff
@@ -265,6 +256,7 @@ class CtaViewModelTest {
 
     @Test
     fun whenCtaDismissedAndUserHasPendingOnboardingCtasThenStageNotCompleted() = coroutineRule.runBlocking {
+        givenControlGroup()
```
[the remark this answers is not in the document]
Comment (cmonfortep) [reply app/src/androidTest/java/com/duckduckgo/app/cta/ui/CtaViewModelTest.kt]: 👍

**Comment by cmonfortep** [comment]
@marcosholgado this is now ready for a second review. Database updates should now be fixed. And there's a small improvement to stop receiving updates from databases when fire pulsing animation is not going to be active again.

File: app/src/main/java/com/duckduckgo/app/cta/ui/CtaViewModel.kt [hunk 4]
```diff
@@ -358,12 +357,35 @@ class CtaViewModel @Inject constructor(
...
+            return@map false
+        }
+
+    @ExperimentalCoroutinesApi
+    private fun getShowFireButtonPulseAnimationFlow(): Flow<Boolean> = dismissedCtaDao.dismissedCtas()
+        .combine(forceStopFireButtonPulseAnimationFlow(), ::Pair)
+        .onEach { (_, forceStopAnimation) ->
+            withContext(dispatchers.io()) {
+                if (pulseAnimationDisabled()) {
+                    println("pulseAnimationDisabled")
```
Comment (marcosholgado) [inline app/src/main/java/com/duckduckgo/app/cta/ui/CtaViewModel.kt:373]: Nit: do we need this print?

File: app/src/main/java/com/duckduckgo/app/cta/ui/CtaViewModel.kt [hunk 5]
```diff
@@ -358,12 +357,35 @@ class CtaViewModel @Inject constructor(
...
+                    dismissPulseAnimation()
+                }
+            }
+        }.shouldShowPulseAnimation()
+
+    @ExperimentalCoroutinesApi
     private fun Flow<Pair<List<DismissedCta>, Boolean>>.shouldShowPulseAnimation(): Flow<Boolean> {
         return this.map { (dismissedCtaDao, forceStopAnimation) ->
             withContext(dispatchers.io()) {
+                println("collectedcta $dismissedCtaDao")
```
Comment (marcosholgado) [inline app/src/main/java/com/duckduckgo/app/cta/ui/CtaViewModel.kt:386]: Nit: do we need this print?

**Review by marcosholgado — APPROVED** [review]
Nice job @cmonfortep! I left a minor comment in a couple of places regarding the println statement, nothing major so feel free to merge!

**Pull request merged**